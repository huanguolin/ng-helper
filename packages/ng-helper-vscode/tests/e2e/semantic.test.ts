import { expect } from 'chai';
import { describe, it, before } from 'mocha';
import * as vscode from 'vscode';

import {
    APP_PAGES_P3_HTML_PATH,
    NUMBER_CHECK_DIRECTIVE_JS_PATH,
    SEMANTIC_TOKENS_COMMAND,
    SEMANTIC_TOKENS_LEGEND_COMMAND,
} from '../testConstants';
import { activate, setupChaiSnapshotPlugin, sleep } from '../testUtils';

describe('Semantic', () => {
    setupChaiSnapshotPlugin();

    before(async () => {
        await activate();
    });

    it('html semantic', async () => {
        await testSemantic(APP_PAGES_P3_HTML_PATH);
    });

    it('inline html semantic', async () => {
        await testSemantic(NUMBER_CHECK_DIRECTIVE_JS_PATH);
    });
});

async function testSemantic(filePath: string, waitSeconds?: number) {
    // show the document
    await vscode.window.showTextDocument(vscode.Uri.file(filePath));
    if (waitSeconds) {
        await sleep(waitSeconds * 1000);
    }

    // get semantic tokens legend
    const semanticTokensLegend = await vscode.commands.executeCommand<vscode.SemanticTokensLegend>(
        SEMANTIC_TOKENS_LEGEND_COMMAND,
        vscode.Uri.file(filePath),
    );

    // get semantic tokens
    const semanticTokens = await vscode.commands.executeCommand<vscode.SemanticTokens>(
        SEMANTIC_TOKENS_COMMAND,
        vscode.Uri.file(filePath),
    );

    // assert
    expect(semanticTokensLegend.tokenModifiers).to.be.empty;
    expect(semanticTokensLegend.tokenTypes).to.be.deep.equals(['string']);
    expect(semanticTokens.data.length).to.be.greaterThan(0);
    expect(semanticTokens).toMatchSnapshot();
}
