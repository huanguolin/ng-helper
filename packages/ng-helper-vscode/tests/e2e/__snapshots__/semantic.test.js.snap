// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Semantic html semantic 1`] = `
Ki {
  "data": Uint32Array [
    3,
    30,
    4,
    0,
    0,
    1,
    17,
    9,
    0,
    0,
    2,
    21,
    2,
    0,
    0,
  ],
  "resultId": undefined,
}
`;

exports[`Semantic inline html semantic 1`] = `
Ki {
  "data": Uint32Array [
    15,
    30,
    4,
    0,
    0,
    1,
    17,
    9,
    0,
    0,
    2,
    21,
    2,
    0,
    0,
  ],
  "resultId": undefined,
}
`;
